### ISSUE TITLE
Search results not working - URGENT !

### ISSUE THREAD

**Issue by rodolphebertozzo** [soflyy/breakdance-bugs#779]
When you create a search results template, and launch a search, the query does not work, you are returned to articles that have nothing to do with the theme of the search on the site

**Comment by rodolphebertozzo** [comment]
@a-dubiel

**Comment by a-dubiel** [comment]
@rodolphebertozzo this is a known bug. Will be addressed in the next stable release (1.3).

**Comment by rodolphebertozzo** [comment]
@a-dubiel status ?

**Comment by louisreingold** [comment]
This was not addressed in 1.3. Much more work needs to be done to even evaluate what we should do here.

**Comment by rodolphebertozzo** [comment]
@louisreingold 
We've known about this problem for a year now, and you've been selling Breakdance as a no-code solution where you can create the most personalized e-business site possible, where you can create visually appealing pages, where you can feed a blog, and you're telling us that fixing this bug that impacts the ergonomics of sites with a blog isn't going to happen right away. I find this unacceptable (you have removed the planned milestone.).

**Comment by rodolphebertozzo** [comment]
@louisreingold is it possible to have an answer?

**Comment by louisreingold** [comment]
We've addressed the search issues in 1.6 - search results are correctly limited by custom post type, search works with WooCo, you can set a custom post type in the Search Form element, search results are compatible with Woo's filtering options, and the Breakdance Search Form element now takes you to a search results page are now compatible with FacetWP and WP Grid Builder filtering of custom post types.

We've also made a11y improvements to the search form.

1.6 is currently in beta, and you can see a demo of the search functionality in action here: https://www.youtube.com/watch?v=NVbPxRuTxPY

Thanks for the report!

**Comment by rodolphebertozzo** [comment]
> We've addressed the search issues in 1.6 - search results are correctly limited by custom post type, search works with WooCo, you can set a custom post type in the Search Form element, search results are compatible with Woo's filtering options, and the Breakdance Search Form element now takes you to a search results page are now compatible with FacetWP and WP Grid Builder filtering of custom post types.
> 
> We've also made a11y improvements to the search form.
> 
> 1.6 is currently in beta, and you can see a demo of the search functionality in action here: https://www.youtube.com/watch?v=NVbPxRuTxPY
> 
> Thanks for the report!

@louisreingold, I assure you that it still doesn't work, look for yourself or maybe it's me who has an error? : https://rodolphebertozzo.fr/?post_type=post&s=base+de+donn%C3%A9es

**Comment by louisreingold** [comment]
1. Make sure there is no custom query set on the Posts List / Post Loop Builder on your search results page.

2. After doing that, if you don't like the results, de-activate Breakdance and then go to the same URL... same results?

**Comment by louisreingold** [comment]
tl;dr: WordPress handles the query on archive and search results page. If you override the query, you'll get exactly what you asked for - an overridden query, which means the post returned will not match your archive or search query.

--------

Basically, if you set a custom query on a Posts List - it overrides the default query.

On a search results page you want to treat it just like an archive and let WP handle the query - since it's WP that is doing the searching behind the scenes. Same goes for any archive.

The reason we allow custom query in Posts List, Post Loop Builder, etc. is because people want to use them on places other than an archive/search results page. But if you are using on an archive or search results page, don't override the query.

(Also, some people want to build a sidebar or footer on an archive page or search results page that shows latest blog posts... for example, check the breakdance.com footer for an example. In that case, it also makes sense to set a custom query and the archive / default query would be ignored.)

**Comment by rodolphebertozzo** [comment]
@louisreingold actually, it's all good ;)
